File: backend/schemas/broker.py
from typing import Optional

from pydantic import BaseModel


class BrokerConfigCreate(BaseModel):
    broker_name: str
    api_key: str
    api_secret: str
    redirect_url: str
    client_id: Optional[str] = None


class BrokerConfigResponse(BaseModel):
    broker_name: str
    api_key_masked: str
    api_secret_masked: str
    redirect_url: str
    is_active: bool
    client_id: Optional[str] = None


class BrokerListItem(BaseModel):
    name: str
    display_name: str
    supported_exchanges: list[str]
    is_configured: bool = False
    is_active: bool = False
    oauth_type: str = ""


class AngelLoginPayload(BaseModel):
    clientcode: str
    broker_pin: str
    totp_code: str

File: frontend/src/types/broker.ts
export interface BrokerListItem {
  name: string;
  display_name: string;
  supported_exchanges: string[];
  is_configured: boolean;
  is_active: boolean;
  oauth_type: string;
}

export interface BrokerConfigData {
  broker: string;
  api_key: string;
  api_secret: string;
  redirect_url: string;
  client_id?: string;
  api_key_market?: string;
  api_secret_market?: string;
}

export interface BrokerConfigResponse {
  status: string;
  message: string;
}

export interface BrokerRedirectResponse {
  url: string;
  kind?: "internal" | "external" | "direct";
}

export interface AngelLoginPayload {
  clientcode: string;
  broker_pin: string;
  totp_code: string;
}

export interface BrokerConnectionStatus {
  connected: boolean;
  token_valid: boolean;
  broker: string | null;
  display_name: string;
  client_id: string | null;
  user_id: string | null;
  trading_client_id: string | null;
  latency_ms: number | null;
  message: string;
  checked_at: string;
}
